### PR TITLE
IGNITE-21811 Cleanup in RocksDbClusterStateStorage

### PR DESCRIPTION
**Pull request by sashapolo**
https://issues.apache.org/jira/browse/IGNITE-21811

Thank you for submitting the pull request.

To streamline the review process of the patch and ensure better code quality
we ask both an author and a reviewer to verify the following:

### The Review Checklist
- [ ] **Formal criteria:** TC status, codestyle, mandatory documentation. Also make sure to complete the following:  
\- There is a single JIRA ticket related to the pull request.  
\- The web-link to the pull request is attached to the JIRA ticket.  
\- The JIRA ticket has the Patch Available state.  
\- The description of the JIRA ticket explains WHAT was made, WHY and HOW.  
\- The pull request title is treated as the final commit message. The following pattern must be used: IGNITE-XXXX Change summary where XXXX - number of JIRA issue.
- [ ] **Design:** new code conforms with the design principles of the components it is added to.
- [ ] **Patch quality:** patch cannot be split into smaller pieces, its size must be reasonable.
- [ ] **Code quality:** code is clean and readable, necessary developer documentation is added if needed.
- [ ] **Tests code quality:** test set covers positive/negative scenarios, happy/edge cases. Tests are effective in terms of execution time and resources.

### Notes
- [Apache Ignite Coding Guidelines](https://cwiki.apache.org/confluence/display/IGNITE/Java+Code+Style+Guide)

### CODE REVIEW

File: modules/cluster-management/src/main/java/org/apache/ignite/internal/cluster/management/raft/RocksDbClusterStateStorage.java [hunk 1]
```diff
@@ -68,33 +71,46 @@ public class RocksDbClusterStateStorage implements ClusterStateStorage {
...
+            RocksDB db = RocksDB.open(options, dbPath.toString());
 
             ColumnFamily defaultCf = ColumnFamily.wrap(db, db.getDefaultColumnFamily());
 
             snapshotManager = new RocksSnapshotManager(db, List.of(fullRange(defaultCf)), snapshotExecutor);
+
+            this.db = db;
         } catch (RocksDBException e) {
-            throw new IgniteInternalException("Failed to start the storage", e);
+            throw new StorageException("Failed to start the storage", e);
```
Comment (tkalkirill) [inline modules/cluster-management/src/main/java/org/apache/ignite/internal/cluster/management/raft/RocksDbClusterStateStorage.java:104]: Why StorageException? IgniteInternalException looks better

File: modules/cluster-management/src/main/java/org/apache/ignite/internal/cluster/management/raft/StorageException.java [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+package org.apache.ignite.internal.cluster.management.raft;
+
+import static org.apache.ignite.lang.ErrorGroups.Common.INTERNAL_ERR;
+
+import org.apache.ignite.internal.lang.IgniteInternalException;
+
+/**
+ * Exception used by {@link RocksDbClusterStateStorage} to signal about errors.
+ */
+public class StorageException extends IgniteInternalException {
```
Comment (tkalkirill) [inline modules/cluster-management/src/main/java/org/apache/ignite/internal/cluster/management/raft/StorageException.java:27]: Please rename to `CmgStorageException`

**Review by tkalkirill — APPROVED** [review]

File: modules/cluster-management/src/main/java/org/apache/ignite/internal/cluster/management/raft/RocksDbClusterStateStorage.java (hunk 1 above)
Comment (sashapolo) [reply modules/cluster-management/src/main/java/org/apache/ignite/internal/cluster/management/raft/RocksDbClusterStateStorage.java:104]: It's deprecated

File: modules/cluster-management/src/main/java/org/apache/ignite/internal/cluster/management/raft/StorageException.java (hunk 2 above)
Comment (sashapolo) [reply modules/cluster-management/src/main/java/org/apache/ignite/internal/cluster/management/raft/StorageException.java:27]: Done

**Review by tkalkirill — APPROVED** [review]

**Pull request merged**